### ISSUE TITLE
A false positive compilation error about trait bounds

### ISSUE THREAD

**Issue by wobniW** [rust-lang/rust#149024]
I tried this code:

```rust
#![feature(const_trait_impl)]
#![feature(generic_const_exprs)]

struct MyStruct<const X: bool>;
const trait MyTrait {
    fn foo() -> bool {
        true
    }
}
const fn func<A: [const] MyTrait>() where MyStruct<{ <A as MyTrait>::foo() }>: Copy {}
```

I expected to see this happen: The code should compile successfully

Instead, this happened: Compiler error:

```
error[E0277]: the trait bound `A: const MyTrait` is not satisfied
  --> src\lib.rs:10:55
   |
10 | const fn func<A: [const] MyTrait>() where MyStruct<{ <A as MyTrait>::foo() }>: Copy {}
   |                                                       ^
```

### Meta

`rustc --version --verbose`:
```
rustc 1.93.0-nightly (2286e5d22 2025-11-13)
binary: rustc
commit-hash: 2286e5d224b3413484cf4f398a9f078487e7b49d
commit-date: 2025-11-13
host: x86_64-pc-windows-gnu
release: 1.93.0-nightly
LLVM version: 21.1.5
```

**Comment by theemathas** [comment]
This is working as intended.

`A: [const] MyTrait` means that "for the const fn to be called in const, `A` must implement `const MyTrait`, but for the const fn to be called at run time only requires `A` to implement `MyTrait`.

Therefore, other code is allowed to call `func` with an `A` type that doesn't implement `const MyTrait`, as long as this call is done at run time. In such a case, the compiler cannot determine the value of the const generic at compile time.

Changing `[const] MyTrait` to `const MyTrait` makes the code compile.
